deeprank/learn/DataSet.py

```diff
@@ -1431,32 +1438,36 @@ def map_feature(self, feat_names, mol_data, grid, npts, angle, axis):
             tmp_feat_ser = [np.zeros(npts), np.zeros(npts)]
             tmp_feat_vect = [np.zeros(npts), np.zeros(npts)]
             data = np.array(mol_data['features/' + name][()])
-
-            chain = data[:, 0]
-            pos = data[:, 1:4]
-            feat_value = data[:, 4]
-
-            if angle is not None:
-                pos = pdb2sql.transform.rot_xyz_around_axis(
-                    pos, axis, angle, center)
-
-            if __vectorize__ or __vectorize__ == 'both':
-
-                for chainID in [0, 1]:
-                    tmp_feat_vect[chainID] = np.sum(
-                        vmap(pos[chain == chainID, :],
-                             feat_value[chain == chainID]),
-                        0)
-
-            if not __vectorize__ or __vectorize__ == 'both':
-
-                for chainID, xyz, val in zip(chain, pos, feat_value):
-                    tmp_feat_ser[int(chainID)] += \
-                        self._featgrid(xyz, val, grid, npts)
-
-            if __vectorize__ == 'both':
-                assert np.allclose(tmp_feat_ser, tmp_feat_vect)
-
+            
+            if data.shape[0] > 0 :
```

Comment:
that's not very clear as a condition. Wouldn't it be better to do right after we create `data`
```python
if data.shape[0]==0:
     logger.warning('Data %s not in the database' %name)
    continue
```

Reply:
I actually don't understand when would `data.shape[0]==0`. What if `name` is not in `data_mol['features']` ?

Reply:
Running Deeprank, I met the following problem:
`name` is in `data_mol['features']` BUT no feature is computed for that feature, then `data` is empty.
This is particularly the case for the RDC* features, where some type of polar contact are absent from the protein protein interface.

The warning should be:
`logger.warning('No %s retrieved at the protein/protein interface' %name)
`